### PR TITLE
Add file upload REST API endpoint

### PR DESCRIPTION
**Pull request by c-st**
### Description

  This PR adds a REST API endpoint for file uploads to the anytype-heart API. The new endpoint exposes the existing
  `FileUpload` RPC functionality via a standard HTTP multipart/form-data interface.

  **Endpoint:** `POST /v1/spaces/{space_id}/files`

  **Features:**
  - Accepts multipart/form-data file uploads with standard `file` field
  - Returns file object metadata including ObjectId and IPFS CID (FileId)
  - Integrates with existing authentication and rate limiting middleware
  - Follows existing API patterns and conventions

  **Implementation details:**
  - Handler receives multipart upload, saves to temp file, and calls FileUpload RPC
  - Service layer wraps RPC call with proper error handling
  - Temporary files are cleaned up automatically after processing
  - Response includes file details as JSON

  ### What type of PR is this? (check all applicable)

  - [x] 🍕 Feature

  ### Related Tickets & Documents

  This adds REST API coverage for file operations, complementing the existing RPC interface. No specific issue ticket was
  provided for this enhancement.

  ### Mobile & Desktop Screenshots/Recordings

  N/A - Backend API change only. Example usage:

  ```bash
  curl -X POST \
    -H "Authorization: Bearer YOUR_TOKEN" \
    -F "file=@/path/to/file.pdf" \
    http://localhost:31006/v1/spaces/SPACE_ID/files

  Example response:
  {
    "object_id": "obj123",
    "file_id": "bafyreiabc123",
    "details": {
      "name": "file.pdf",
      "sizeInBytes": 1024
    }
  }
```

### Added tests?

- [x] 👍 yes

Test coverage includes:
  - Handler tests (3 test cases): successful upload, error handling, edge cases
  - Service tests (3 test cases): RPC response handling, error mapping, nil details
  - All tests follow existing AAA (Given/When/Then) pattern
  - Tests use proper mocking of dependencies

### Added to documentation?

- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by c-st** [comment]
Hey @deff7. Done! I've rebased onto develop and resolved the merge conflicts with the recent API refactor. The main change was adapting the file upload route registration to use the new registerFileRoutes() pattern that was introduced.

Tests and build are passing.

File: core/api/service/file.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+var (
+	ErrFailedUploadFile = errors.New("failed to upload file")
+)
+
+// UploadFile uploads a file to the specified space
+func (s *Service) UploadFile(ctx context.Context, spaceId string, localPath string) (*apimodel.FileUploadResponse, error) {
+	req := &pb.RpcFileUploadRequest{
+		SpaceId:   spaceId,
+		LocalPath: localPath,
+		Type:      model.BlockContentFile_File, // default to generic file
```
Comment (deff7) [inline core/api/service/file.go:24]: You should use default (None) type here. The file uploader will automatically detect proper file type

**Review by deff7 — CHANGES_REQUESTED** [review]

**Comment by martinholovsky** [comment]
Adding a user story in case it helps prioritisation. I'm building a browser extension (Brave/Chrome) that clips web articles into an Anytype space as Markdown pages — the analogue of Evernote/Notion Web Clipper. Today I can only hot-link images back to their origin, so clips silently break when source pages change or go behind auth. This endpoint is the last missing piece for a self-contained clip.

The PR looks ready — rebased onto `develop`, CLA signed, tests/build passing since December. Any chance a maintainer could take another review pass?

**Comment by Axl-Lvy** [comment]
Hi, could we by any chance look at this dev please ? I want this feature so much
